### PR TITLE
Add api references

### PR DESCRIPTION
**Pull request by Yisaer**
### Which TiDB Operator version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v1.1 (TiDB Operator 1.1 versions)
- [ ] v1.0 (TiDB Operator 1.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: 
- Other reference link(s):

### CODE REVIEW

File: zh/TOC.md [hunk 1]
```diff
@@ -67,3 +67,4 @@
   - [增强型 StatefulSet 控制器](advanced-statefulset.md)
 - [故障诊断](troubleshoot.md)
 - [常见问题](faq.md)
+- [Operator API 参考文档](api-references.md)
```
Comment (DanielZhangQD) [inline zh/TOC.md:70]:
```suggestion
- [TiDB Operator API 参考文档](api-references.md)
```

**Comment by ran-huang** [comment]
@Yisaer Could you please also add the files to the `en` directory so our English users can see them?

**Comment by Yisaer** [comment]
@ran-huang   PTAL

File: en/api-references.md [hunk 2]
```diff
@@ -0,0 +1,12981 @@
+---
+title: TiDB Operator API Document
+summary: Reference of TiDB Operator API
+category: how-to
```
Comment (ran-huang) [inline en/api-references.md:4]:
```suggestion
category: reference
```

**Pull request merged**